packages/router/router.ts

```diff
@@ -437,8 +437,10 @@ type LinkNavigateOptions = BaseNavigateOptions;
 type SubmissionNavigateOptions = BaseNavigateOptions & {
   formMethod?: HTMLFormMethod;
   formEncType?: FormEncType;
-  formData: FormData;
-};
+} & (
+    | { formData: FormData; payload?: undefined }
+    | { formData?: undefined; payload: any }
```

Comment:
`formData`/`payload` are mutually exclusive on the `navigation`- you can only include one in a call to `router.navigate`.  But they are _not_ mutually exclusive in the `action` since you could have sent a `payload` but provided an `encType` that resulted in the router serializing the `payload` into the `request`